cms/djangoapps/contentstore/tests/test_i18n.py

```diff
@@ -4,6 +4,129 @@
 
 from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
 from contentstore.tests.utils import AjaxEnabledTestClient
+from xmodule.modulestore.django import ModuleI18nService
+from django.utils import translation
+from django.utils.translation import get_language
+import mock
+import gettext
+
+
+class FakeTranslations(ModuleI18nService):
+    """A test GNUTranslations class that takes a map of msg -> translations."""
+
+    def __init__(self, translations):  # pylint: disable=super-init-not-called
+        self.translations = translations
+
+    def ugettext(self, msgid):
+        return self.translations.get(msgid, msgid)
+
+    @staticmethod
+    def translator(locales_map):  # pylint: disable=method-hidden
+        """Build mock translator for the given locales.
+        Returns a mock gettext.translation function that uses
+        individual TestTranslations to translate in the given locales.
+        :param locales_map: A map from locale name to a translations map.
+                            {
+                             'es': {'Hi': 'Hola', 'Bye': 'Adios'},
+                             'zh': {'Hi': 'Ni Hao', 'Bye': 'Zaijian'}
+                            }
+        """
+        def _translation(domain, localedir=None, languages=None):  # pylint: disable=unused-argument
+            """
+            return gettext.translation for given language
+            """
+            if languages:
+                language = languages[0]
+                if language in locales_map:
+                    return FakeTranslations(locales_map[language])
+            return gettext.NullTranslations()
+        return _translation
+
+
+class TestModuleI18nService(ModuleStoreTestCase):
+    """ Test ModuleI18nService """
+
+    xblock_name = 'dummy_block'
+
+    def setUp(self):
+        """ Setting up tests """
+        super(TestModuleI18nService, self).setUp()
+        self.test_language = 'dummy language'
+        self.xblock_mock = mock.Mock()
+        self.xblock_mock.unmixed_class.__name__ = self.xblock_name
+        self.i18n_service = ModuleI18nService(self.xblock_mock)
+
+    def test_django_service_translation_works(self):
+        """
+        Test django translation service works fine.
+        """
+
+        def wrap_with_xyz(func):
+            """
+            A decorator function that just adds 'XYZ ' to the front of all strings
+            """
+            def new_func(*args, **kwargs):
+                """ custom function """
+                output = func(*args, **kwargs)
+                return "XYZ " + output
+            return new_func
+
+        old_lang = translation.get_language()
+
+        # Activate french, so that if the fr files haven't been loaded, they will be loaded now.
+        translation.activate("fr")
```

Comment:
If the test is going to change global state like the current language, then you need an `addCleanup` in setUp that will restore it.  Setting it back inside the test isn't enough because the test won't complete if it fails.